### PR TITLE
Add "Manage Recalls" system (PPUD Replacement)

### PR DESCRIPTION
**Pull request by mr-ken**
## What does this pull request do?

Adds "Manage Recalls" system services (PPUD Replacement project).

## What is the intent behind these changes?

To contribute to the HMPPS Digital architecture diagrams.

This is my first attempt at updating this project ; comments/suggestion highly welcome. Don't be shy!

### CODE REVIEW

File: src/main/kotlin/model/ManageRecalls.kt [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        "Manage Recalls API Database",
+        "Database to store recalls caseworking info",
+        "RDS Postgres DB"
+      ).apply {
+        Tags.DATABASE.addTo(this)
+        CloudPlatform.rds.add(this)
+      }
+
+      prisonsRegisterContainer = system.addContainer(
+        "Prisons Register",
```
Comment (sldblog) [inline src/main/kotlin/model/ManageRecalls.kt:41]: 🙋‍♂️ Probably worth adding as a separate thing, as we have [CourtRegister](https://github.com/ministryofjustice/hmpps-architecture-as-code/blob/17570ed350ea7bbdedb665d98325c95c6b68611c/src/main/kotlin/model/CourtRegister.kt) too

File: src/main/kotlin/model/ManageRecalls.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        uses(NOMIS.offenderSearch, "searches for prisoners")
+        uses(manageRecallsApiContainer, "operates on", "HTTPS")
+        uses(prisonsRegisterContainer, "gets prisons", "HTTPS")
+        CloudPlatform.kubernetes.add(this)
+        CloudPlatform.elasticache.add(this)
+      }
+
+      ppcsCaseWorker = model.addPerson("PPCS caseworker", "They case manage recalls")
+      ppcsCaseWorker.uses(manageRecallsUiContainer, "visits", "HTTPS")
+      ppcsCaseWorker.uses(HMPPSAuth.system, "Login")
```
Comment (sldblog) [inline src/main/kotlin/model/ManageRecalls.kt:77]: Relationships to things outside this class should be declared in `defineRelationships` (which removes the need to put the new element at the bottom of `MODEL_ITEMS`)

File: src/main/kotlin/model/ManageRecalls.kt [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      }
+
+      manageRecallsUiContainer = system.addContainer(
+        "Manage Recalls Web Application",
+        "Web application for the case management of recalls",
+        "Node Express app"
+      ).apply {
+        setUrl("https://github.com/ministryofjustice/manage-recalls-ui")
+        Tags.WEB_BROWSER.addTo(this)
+        uses(NOMIS.offenderSearch, "searches for prisoners")
```
Comment (sldblog) [inline src/main/kotlin/model/ManageRecalls.kt:68]: Relationships to things outside this class should be declared in `defineRelationships` (which removes the need to put the new element at the bottom of `MODEL_ITEMS`)

**Review by sldblog — COMMENTED** [review]
Great stuff, just one comment on adding relationships to other systems/containers to another function

File: src/main/kotlin/model/ManageRecalls.kt (hunk 2 above)
Comment (sldblog) [reply src/main/kotlin/model/ManageRecalls.kt:77]: Come to think of it, `defineRelationshipsToOtherSystems` might be a better name

File: src/main/kotlin/model/ManageRecalls.kt (hunk 1 above)
Comment (mr-ken) [reply src/main/kotlin/model/ManageRecalls.kt:41]: gotcha - thanks for the nudge!

File: src/main/kotlin/model/ManageRecalls.kt (hunk 2 above)
Comment (mr-ken) [reply src/main/kotlin/model/ManageRecalls.kt:77]: ok - changing

File: src/main/kotlin/model/ManageRecalls.kt (hunk 3 above)
Comment (mr-ken) [reply src/main/kotlin/model/ManageRecalls.kt:68]: ok

**Review by sldblog — APPROVED** [review]
Thank you so much!

**Pull request merged**